toh-oo.rb

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/ruby
+# Towers of Hanoi\
+# To run this enter the following, create a new TOH object
+# e.g. tow = TowerOfHanoi.new(3)
+# then run the play function
+# tow.play
+
+# Stores the player's inputs
+class Player
+  attr_accessor :user_input
+end
+
+
+class Board
+
+  attr_accessor :toh
+
+  def initialize(n)
+    @toh = Array.new(3)
+    @toh[0] = (1..n).to_a
+    @toh[1], @toh[2] = [],[]
+  end
+
+  # Draw the current state of the board
+  def render
+
+    # Loop over each pin array.
+    @toh.each do |pin|
+
+      # Loop over each disk on the pins.
+
+      if pin  
+        pin.each do |disk|
+          puts disk
+        end
+      end
+
+    # End of pin.
+    puts "--"
+    end
+  end
+end
+
+class TowerOfHanoi
+
+  # Intialise the board
+  def initialize(n)
+    @no_of_disks = n
+    @won = false
+    @game_play = true
+    @player = Player.new
+    @board = Board.new(@no_of_disks)
+  end
+
+  def play
+    puts "Welcome to Tower of Hanoi!"
+    puts "Instructions:"
+    puts "Enter where you'd like to move from and to"
+    puts "in the format '1,3'. Enter 'q' to quit."
+
+    while (@won != true && @game_play != false)   
```

Comment:
Can be rewritten as `until (@won || !@game_play)` or `while(!@won && @game_play)